src/main/kotlin/com/jaoafa/vcspeaker/tts/providers/SpeechProvider.kt

```diff
@@ -0,0 +1,75 @@
+package com.jaoafa.vcspeaker.tts.providers
+
+import com.jaoafa.vcspeaker.tts.providers.soundmoji.SoundmojiContext
+import com.jaoafa.vcspeaker.tts.providers.soundmoji.SoundmojiProvider
+import com.jaoafa.vcspeaker.tts.providers.voicetext.VoiceTextProvider
+import com.jaoafa.vcspeaker.tts.providers.voicetext.VoiceTextContext
+
+/**
+ * 音声データを提供するプロバイダーを表すインターフェースです。
+ *
+ * @param T [ProviderContext] の型
+ * @property id プロバイダー ID; 一意である必要があります。
+ * @property format 音声データのフォーマット
+ */
+interface SpeechProvider<T : ProviderContext> {
+    val id: String
+    val format: String
+
+    /**
+     * 与えられた [ProviderContext] に対応する音声データを提供します。
+     *
+     * @param context [ProviderContext] のインスタンス
+     * @return 音声データ
+     */
+    suspend fun provide(context: T): ByteArray
+}
+
+/**
+ * 音声データの生成・取得に必要な情報を表すインターフェースです。
+ * ProviderContext によって取得される音声は、必ず一意である必要があります。(Cache に使用するため)
```

Comment:
音声データがキャッシュ(のキー？)に使用されるわけではないですよね…？
音声の一意性って何…？hashが一意であるべきということ？

Reply:
ちょっと冗長かもしれませんが、これでどうでしょう

```suggestion
 * 音声はキャッシュされるため、ある ProviderContext に対して取得される音声は、必ず一意に定まる必要があります。同一の ProviderContext のときに、異なる実行によって異なる音声が出力されてはいけません。
```

Reply:
適用します